common/djangoapps/student/helpers.py

```diff
@@ -127,11 +143,17 @@ def check_verify_status_by_course(user, course_enrollments):
             )
             if status is None and not submitted:
                 if deadline is None or deadline > datetime.now(UTC):
-                    if has_active_or_pending:
-                        # The user has an active verification, but the verification
-                        # is set to expire before the deadline.  Tell the student
-                        # to reverify.
-                        status = VERIFY_STATUS_NEED_TO_REVERIFY
+                    if SoftwareSecurePhotoVerification.user_is_verified(user):
+                        if SoftwareSecurePhotoVerification.is_verification_expiring_soon(expiration_datetime):
+                            # The user has an active verification, but the verification
+                            # is set to expire within "VERIFICATION_EXPIRATION_DAYS" days (default is 4 weeks).
+                            # Tell the student to reverify.
+                            status = VERIFY_STATUS_NEED_TO_REVERIFY
+                        else:
+                            if verification_expiring_soon:
+                                status = VERIFY_STATUS_RESUBMITTED
```

Comment:
This part looks strange to me. `verification_expiring_soon = SoftwareSecurePhotoVerification.is_verification_expiring_soon(expiration_datetime)`. So why bother checking `verification_expiring_soon` after checking `SoftwareSecurePhotoVerification.is_verification_expiring_soon(expiration_datetime)`? In fact, why not just use `verification_expiring_soon` on line 147?